### PR TITLE
Cache `pr-filters` API request

### PR DESCRIPTION
**Pull request by FloEdelmann**
Follow-up to #2757, checks one item from #2671.

* Cache CI status for faster Status filter
* Rename Status → Checks (as mentioned in https://github.com/sindresorhus/refined-github/pull/2757#issuecomment-583701763)
* Call API instead of relying on `ci-link`
* Fix some spacing again, because GitHub likes to change stuff around

BTW (in reply to https://github.com/sindresorhus/refined-github/pull/2757#issuecomment-583273593), the `status:pending` filter works again as expected :tada: 

# Test

(same links as in the original PR #2133)

* https://github.com/sindresorhus/refined-github/pulls (Filter should be present)
* https://github.com/HardikModha/github-playground/pulls (Filters should not be present)

Screenshot_2020-02-13_17-00-46

### CODE REVIEW

File: source/features/pr-filters.tsx [hunk 1]
```diff
@@ -60,43 +62,63 @@ function addDraftFilter({delegateTarget: reviewsFilter}: DelegateEvent): void {
...
+				history(first: 10) {
+					nodes {
+						status {
+							state
+						}
+					}
+				}
+			}
+		}
+	}`);
```
Comment (fregante) [inline source/features/pr-filters.tsx]: You can indent like this
```js
	const {repository} = await api.v4(`
		repository(${getRepoGQL()}) {
			head: object(expression: "HEAD") {
				... on Commit {
					history(first: 10) {
						nodes {
							status {
								state
							}
						}
					}
				}
			}
		}
	`);
```

File: source/features/pr-filters.tsx [hunk 2]
```diff
@@ -60,43 +62,63 @@ function addDraftFilter({delegateTarget: reviewsFilter}: DelegateEvent): void {
 	addDropdownItem(dropdown, 'Not ready for review (Draft PR)', 'draft', 'true');
 }
 
-async function addStatusFilter(): Promise<void> {
+const cachedChecksStatus = cache.function(async () => {
```
Comment (fregante) [inline source/features/pr-filters.tsx]:
```suggestion
const hasChecks = cache.function(async (): boolean => {
```

File: source/features/pr-filters.tsx [hunk 3]
```diff
@@ -60,43 +62,63 @@ function addDraftFilter({delegateTarget: reviewsFilter}: DelegateEvent): void {
...
+async function addChecksFilter(): Promise<void> {
 	const reviewsFilter = select(reviewsFilterSelector);
 	if (!reviewsFilter) {
 		return;
 	}
 
-	// TODO: replace this with an API call
-	const hasCI = await fetchCIStatus();
+	const hasCI = await cachedChecksStatus();
 	if (!hasCI) {
```
Comment (fregante) [inline source/features/pr-filters.tsx]:
```suggestion
	if (!await hasChecks()) {
```

File: source/features/pr-filters.tsx [hunk 4]
```diff
@@ -60,43 +62,63 @@ function addDraftFilter({delegateTarget: reviewsFilter}: DelegateEvent): void {
...
 function init(): void {
 	delegate(reviewsFilterSelector, 'toggle', addDraftFilter, true);
-	addStatusFilter();
+	addChecksFilter();
 }
 
 features.add({
 	id: __featureName__,
-	description: 'Adds `Build status` and draft PR dropdown filters in PR lists.',
+	description: 'Adds Checks and draft PR dropdown filters in PR lists.',
```
Comment (fregante) [inline source/features/pr-filters.tsx]:
```suggestion
	description: 'Adds Checks and Draft PR dropdown filters in PR lists.',
```

The readme and screenshot should also be updated (arrow not necessary, as long as the Checks dropdown is open)

**Comment by fregante** [comment]
The next step would be to use `element-ready` and `nowAndOnAjaxedPages`. Would you like to extend this PR or should I merge this for now?

**Comment by fregante** [comment]
I'm merging this for now so it appears in today's release

**Comment by fregante** [comment]
I'm merging this for now so it appears in today's release

**Comment by FloEdelmann** [comment]
> The next step would be to use `element-ready` and `nowAndOnAjaxedPages`.

@fregante I will give it a shot when I have some time, unless anybody else has done it already until then.

**Pull request merged**